homeassistant/components/overkiz/climate_entities/somfy_thermostat.py

```diff
@@ -0,0 +1,176 @@
+"""Support for TaHoma Smart Thermostat."""
+from __future__ import annotations
+
+from typing import Any, cast
+
+from pyoverkiz.enums import OverkizCommand, OverkizState
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_HEAT,
+    PRESET_AWAY,
+    PRESET_HOME,
+    PRESET_NONE,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+from ..coordinator import OverkizDataUpdateCoordinator
+from ..entity import OverkizEntity
+
+PRESET_FREEZE = "freeze"
+PRESET_NIGHT = "night"
+
+STATE_DEROGATION_FURTHER_NOTICE = "further_notice"
+STATE_DEROGATION_ACTIVE = "active"
+STATE_DEROGATION_INACTIVE = "inactive"
+STATE_PRESET_AT_HOME = "atHomeMode"
+STATE_PRESET_AWAY = "awayMode"
+STATE_PRESET_FREEZE = "freezeMode"
+STATE_PRESET_MANUAL = "manualMode"
+STATE_PRESET_SLEEPING_MODE = "sleepingMode"
+STATE_PRESET_SUDDEN_DROP_MODE = "suddenDropMode"
+
+OVERKIZ_TO_HVAC_MODES: dict[str, str] = {
+    STATE_DEROGATION_ACTIVE: HVAC_MODE_HEAT,
+    STATE_DEROGATION_INACTIVE: HVAC_MODE_AUTO,
+}
+OVERKIZ_TO_PRESET_MODES: dict[str, str] = {
+    STATE_PRESET_AT_HOME: PRESET_HOME,
+    STATE_PRESET_AWAY: PRESET_AWAY,
+    STATE_PRESET_FREEZE: PRESET_FREEZE,
+    STATE_PRESET_MANUAL: PRESET_NONE,
+    STATE_PRESET_SLEEPING_MODE: PRESET_NIGHT,
+    STATE_PRESET_SUDDEN_DROP_MODE: PRESET_NONE,
+}
+PRESET_MODES_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_PRESET_MODES.items()}
+TARGET_TEMP_TO_OVERKIZ = {
+    PRESET_HOME: "somfythermostat:AtHomeTargetTemperatureState",
+    PRESET_AWAY: "somfythermostat:AwayModeTargetTemperatureState",
+    PRESET_FREEZE: "somfythermostat:FreezeModeTargetTemperatureState",
+    PRESET_NIGHT: "somfythermostat:SleepingModeTargetTemperatureState",
```

Comment:
```suggestion
    PRESET_HOME: OverkizState.SOMFY_THERMOSTAT_AT_HOME_TARGET_TEMPERATURE ,
    PRESET_AWAY: OverkizState.SOMFY_THERMOSTAT_AWAY_MODE_TARGET_TEMPERATURE,
    PRESET_FREEZE: OverkizState.SOMFY_THERMOSTAT_FREEZE_MODE_TARGET_TEMPERATURE,
    PRESET_NIGHT: OverkizState.SOMFY_THERMOSTAT_SLEEPING_MODE_TARGET_TEMPERATURE,
```

Would be good to use the constants.